ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -570,4 +590,138 @@ private void reportUnrecognizedDirs() {
+      File dir = new File(heathcheckDir);
+      if (!dir.exists()) dir.mkdirs();
+
+      //Creates and opens the file
+      AsynchronousFileChannel fileChannel;
+      try {
+        Path path = Paths.get(heathcheckDir,"temp");
+        fileChannel =
+            AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.READ,
+                StandardOpenOption.CREATE, StandardOpenOption.DELETE_ON_CLOSE);
+        fileChannel.force(true); //writes to disk immediately
+      }catch(IOException e){
+        diskHealthStatus = DiskHealthStatus.CREATE_IO_EXCEPTION;
+        logger.error("Exception occurred when creating a file for disk healthcheck", e);
+        return;
+      }
+
+      //Finds today's date and stores it
+      String todayDate = new SimpleDateFormat("yyyy-mm-dd hh:mm:ss").format(Calendar.getInstance().getTime());
+      ByteBuffer buffer = ByteBuffer.allocate(todayDate.getBytes().length);
+      buffer.put(todayDate.getBytes());
+      buffer.flip();
+
+      //Write to file in operationTimeoutSeconds seconds
+      Future<Integer> writeOperation = fileChannel.write(buffer, 0);
+      try{
+        writeOperation.get(operationTimeoutSeconds,TimeUnit.SECONDS);
+      }catch(TimeoutException e){
+        writeOperation.cancel(true);
+        diskHealthStatus = DiskHealthStatus.WRITE_TIMEOUT;
+        logger.error("Timeout Exception occurred when writing to a file for disk healthcheck", e);
+        deleteFile(fileChannel);
+        return;
+      }catch(Exception e){
+        writeOperation.cancel(true);
+        diskHealthStatus = DiskHealthStatus.WRITE_EXCEPTION;
+        logger.error("Exception occurred when writing to a file for disk healthcheck", e);
+        deleteFile(fileChannel);
+        return;
+      }
+
+      //Reads from the file in operationTimeoutSeconds seconds
+      buffer.clear();
+      Future<Integer> readOperation = fileChannel.read(buffer, 0);
+
+      try{
+        readOperation.get(operationTimeoutSeconds,TimeUnit.SECONDS);
+      }catch(TimeoutException e){
+        readOperation.cancel(true);
+        diskHealthStatus = DiskHealthStatus.READ_TIMEOUT;
+        logger.error("Timeout Exception occurred when reading from a file for disk healthcheck", e);
+        deleteFile(fileChannel);
+        return;
+      }catch(Exception e){
+        readOperation.cancel(true);
+        diskHealthStatus = DiskHealthStatus.READ_EXCEPTION;
+        logger.error("Exception occurred when reading from a file for disk healthcheck", e);
+        deleteFile(fileChannel);
+        return;
+      }
+
+      //Checks if the content matches what was written
+      String fileContent = new String(buffer.array()).trim();
+      if(!fileContent.equals(todayDate)){
+        diskHealthStatus = DiskHealthStatus.READ_DIFFERENT;
+        deleteFile(fileChannel);
+        return;
+      }
+
+      //Deletes this test file
+      if(deleteFile(fileChannel)){
```

Comment:
This delete also should be in a try-catch since any disk related operation could fail at any time

Reply:
Yeah so the deleteFile function is in a try catch